src/switch/switch.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import {
+  Switch as NativeSwitch,
+  StyleProp,
+  ViewStyle,
+  ViewProps,
+  Platform,
+} from 'react-native';
+import { withTheme } from '../config';
+import { Theme } from '../config/theme';
+
+export type SwitchProps = {
+  disabled?: boolean;
+  value?: boolean;
+  onValueChange?(value: boolean): void;
+  color?: string;
+  style?: StyleProp<ViewStyle>;
+  theme?: Theme;
+} & ViewProps;
+
+const Switch: React.FunctionComponent<SwitchProps> = ({
+  value,
+  disabled,
+  onValueChange,
+  color,
+  style,
+  theme,
+  ...rest
+}) => {
+  // switchedOnColor deals with picking up a color provided as props by user or picks up default theme
+  const switchedOnColor = color || theme.colors.primary;
+
+  const onTintColor =
+    Platform.OS === 'ios'
+      ? switchedOnColor
+      : disabled
+      ? theme.colors.disabled
+      : switchedOnColor;
+
+  const thumbTintColor =
+    Platform.OS === 'ios'
+      ? undefined
+      : disabled
+      ? theme.colors.disabled
```

Comment:
I don't understand this. If there's a `theme.colors.disabled` color, then choose between `grey1` or `grey4`?

Reply:
That's where my confusion of color lied for the users to use it, if you find irrelevant its okay to remove it. Thanks.